nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -720,11 +748,13 @@ public Document copy(Document source, Document parent, String name) {
         return getDocument(copyState);
     }
 
-    protected String copyRecurse(String sourceId, String parentId, LinkedList<String> ancestorIds, String name) {
+    protected String copyRecurse(String sourceId, String parentId, LinkedList<String> ancestorIds, String name,
+            boolean excludeSpecialChildren, boolean excludeRegularChildren) {
         String copyId = copy(sourceId, parentId, ancestorIds, name);
         ancestorIds.addLast(copyId);
-        for (String childId : getChildrenIds(sourceId)) {
-            copyRecurse(childId, copyId, ancestorIds, null);
+        for (String childId : getChildrenIds(sourceId, excludeSpecialChildren, excludeRegularChildren)) {
+            // don't exclude regular children when recursing
+            copyRecurse(childId, copyId, ancestorIds, null, true, false);
```

Comment:
Let's keep passing `excludeSpecialChildren` instead of `true` to be consistent with VCS.